python/lsst/ctrl/provenance/dc3.py

```diff
@@ -155,7 +159,7 @@ def queryRunOffset(self):
         self._globalDb.condParamString("runId", self._runid)
         self._globalDb.setQueryWhere("runId = :runId")
         self._globalDb.query()
-        if not self._globalDb.next() or self._globalDb.columnIsNull(0):
+        if not next(self._globalDb) or self._globalDb.columnIsNull(0):
```

Comment:
I don't see any tests where we test `next` as the specific tests are disabled in `daf_persistence`. @n8pease it looks like my fix for DM-7226 might be wrong. I am thinking that we need to trap for the final item and then raise `StopIteration` if we want to be compliant with the Python expectation.